pkg/utils/util.go

```diff
@@ -977,58 +977,120 @@ func FormatAndMount(diskMounter *k8smount.SafeFormatAndMount, source string, tar
 			return err
 		}
 		if existingFormat == "" {
-			if readOnly {
-				// Don't attempt to format if mounting as readonly, return an error to reflect this.
-				return errors.New("failed to mount unformatted volume as read only")
-			}
-
-			// Disk is unformatted so format it.
-			args := []string{source}
-			// Use 'ext4' as the default
-			if len(fstype) == 0 {
-				fstype = "ext4"
-			}
-
-			if fstype == "ext4" || fstype == "ext3" {
-				args = []string{
-					"-F",  // Force flag
-					"-m0", // Zero blocks reserved for super-user
-					source,
-				}
-				// add mkfs options
-				if len(mkfsOptions) != 0 {
-					args = []string{}
-					for _, opts := range mkfsOptions {
-						args = append(args, opts)
-					}
-					args = append(args, source)
-				}
-			}
-			log.Infof("Disk %q appears to be unformatted, attempting to format as type: %q with options: %v", source, fstype, args)
-			startT := time.Now()
-
-			pvName := filepath.Base(source)
-			_, err := diskMounter.Exec.Command("mkfs."+fstype, args...).CombinedOutput()
-			log.Infof("Disk format finished, pvName: %s elapsedTime: %+v ms", pvName, time.Now().Sub(startT).Milliseconds())
-			if err == nil {
-				// the disk has been formatted successfully try to mount it again.
-				return diskMounter.Interface.Mount(source, target, fstype, mountOptions)
-			}
-			log.Errorf("format of disk %q failed: type:(%q) target:(%q) options:(%q) error:(%v)", source, fstype, target, args, err)
-			return err
+			return FormatNewDisk(readOnly, source, fstype, target, mkfsOptions, mountOptions, diskMounter)
 		}
 		// Disk is already formatted and failed to mount
 		if len(fstype) == 0 || fstype == existingFormat {
 			// This is mount error
 			return mountErr
 		}
+		// special logic for encrypted disk
```

Comment:
Detect if an encrypted disk is empty disk, since atari type partition is detected by blkid.